### PR TITLE
Add additional support for mapping collections of MappableObjects

### PR DESCRIPTION
**Pull request by justbaum30**
### Description
I have a need on my current project to be able to map many different kinds of data structures. As it stands today, Genome only supports mapping properties if they are in the following data structures:
- `MappableObject`
- `Array<MappableObject>`

This pull request adds support for several additional types of data structures that I believe would be helpful in many scenarios:
- `Array<Array<MappableObject>>`
- `Dictionary<String, MappableObject>`
- `Dictionary<String, Array<MappableObject>>`
- `Set<MappableObject>`

I attempted to follow the formatting of the library as best as I could. There are some scenarios where my code could have been more succinct, but I opted for readability and performance (avoiding duplicate "enforcement" checks) over brevity. Feel free to modify it as you feel necessary.

To ensure no breakage, I created unit tests to cover the code I added and tried to beef up some existing coverage a bit. In particular, I did a lot of work in `SettableOperatorTest` and `FromJsonOperatorTest`. I also took the liberty of deleting several unit tests that weren't being used or weren't providing much value (such as the performance measures).

Let me know if there is anything you want added, removed, or otherwise modified. Thanks for the awesome library and the timely reviews!

### CODE REVIEW

**Comment by justbaum30** [comment]
@LoganWright Do you happen to have any timeline on this? Not trying to rush or pester but unfortunately our team can't start getting approval for the library (big corporate policies) until these changes are in. Thanks!

**Comment by LoganWright** [comment]
@justbaum30 Got a little backed up this week, I'll give it a review by tomorrow!

File: Genome.playground/Sources/SettableOperator.swift [hunk 1]
```diff
@@ -56,6 +80,48 @@ public prefix func <~ <T: MappableObject>(map: Map) throws -> [T] {
...
+public prefix func <~ <T: MappableObject>(map: Map) throws -> [[T]] {
+    try enforceMapType(map, expectedType: .FromJson)
+    let jsonArrayOfArrays = try expectJsonArrayOfArraysWithMap(map, targetType: [[T]].self)
+    
+    var mappedArrayOfArrays: [[T]] = []
+    for jsonArray in jsonArrayOfArrays {
+        let mappedArray = try [T].mappedInstance(jsonArray, context: map.context)
+        mappedArrayOfArrays.append(mappedArray)
+    }
+    return mappedArrayOfArrays
```
Comment (LoganWright) [inline Genome.playground/Sources/SettableOperator.swift]: Let's use a map for this one:

```
    return try jsonArrayOfArrays.map { try [T].mappedInstance($0, context: map.context) }
```

**Comment by LoganWright** [comment]
Hey @justbaum30,

Just got done going through this, and it looks AWESOME!

Thanks for taking the time to really go through and match the style that had been setup in the project.  Also, major props to the testing cleanup!  It was in need of an update, and I really appreciate it.

I had one minor comment about map on one function, but I'm going to merge this anyways and I'll clean that up later since its so minor.

Thanks,

Logan

**Pull request merged**